paasta_tools/setup_marathon_job.py

```diff
@@ -341,6 +368,18 @@ def get_old_happy_unhappy_draining_tasks(other_apps, drain_method, service, nerv
     return old_app_live_happy_tasks, old_app_live_unhappy_tasks, old_app_draining_tasks
 
 
+def undrain_tasks(to_undrain, leave_draining, drain_method, log_deploy_error):
+    # If any tasks on the new app happen to be draining (e.g. someone reverts to an older version with
+    # `paasta mark-for-deployment`), then we should undrain them.
+    for task in to_undrain:
+        if task not in leave_draining:
+            # TODO: don't take actions in deploy_service.
```

Comment:
Yeah, I'd still like to avoid this, but it would require a more extensive refactor -- passing the list of things to drain/undrain into do_bounce. For the sake of expediency I'm going to avoid that for now.

I am, however, going to move this comment to deploy_service where we call undrain_tasks.